### PR TITLE
[MIG][12.0] mgmtsystem

### PR DESCRIPTION
**Pull request by simahawk**
maybe some old commits to squash together...

### CODE REVIEW

File: mgmtsystem/models/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+from . import (
```
Comment (i-vyshnevska) [inline mgmtsystem/models/__init__.py]: lets go to one style

File: mgmtsystem/views/menus.xml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <menuitem
+      id="menu_mgmtsystem_root"
+      name="Management System"
+      web_icon="mgmtsystem,static/description/icon.png"
+      sequence="150"
+      groups="mgmtsystem.group_mgmtsystem_manager,group_mgmtsystem_viewer"
```
Comment (i-vyshnevska) [inline mgmtsystem/views/menus.xml]:
```suggestion
      groups="group_mgmtsystem_manager,group_mgmtsystem_viewer"
```

File: mgmtsystem/views/mgmtsystem_system.xml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      <field name="view_type">form</field>
+      <field name="view_mode">tree,form</field>
+      <field name="view_id" ref="system_tree"/>
+    </record>
+
+    <menuitem
+      id="menu_mgmtsystem_configuration_system"
+      name="Systems"
+      parent="mgmtsystem.menu_mgmtsystem_configuration"
+      groups="mgmtsystem.group_mgmtsystem_manager"
```
Comment (i-vyshnevska) [inline mgmtsystem/views/mgmtsystem_system.xml]:
```suggestion
      groups="group_mgmtsystem_manager"
```

File: mgmtsystem/views/res_config.xml [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+        <field name="model">res.config.settings</field>
+        <field name="inherit_id" ref="base.res_config_settings_view_form"/>
+        <field name="priority" eval="95"/>
+        <field name="arch" type="xml">
+
+            <xpath expr="//div[hasclass('settings')]" position="inside">
+                <div class="app_settings_block" 
+                     data-string="Management System" string="Management System"
+                     data-key="mgmtsystem"
+                     groups="mgmtsystem.group_mgmtsystem_user_manager">
```
Comment (i-vyshnevska) [inline mgmtsystem/views/res_config.xml:15]:
```suggestion
                     groups="group_mgmtsystem_user_manager">
```

File: mgmtsystem/security/mgmtsystem_security.xml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        <field name="name">Viewer</field>
+        <field name="comment">Module user with general read only access.</field>
+        <field name="category_id" ref="module_category_management_system"/>
+    </record>
+
+    <record id="group_mgmtsystem_user" model="res.groups">
+        <field name="name">User</field>
+        <field name="comment">Module user, with general read and write.</field>
+        <field name="category_id" ref="module_category_management_system"/>
+        <field name="implied_ids" eval="[(4,ref('mgmtsystem.group_mgmtsystem_viewer'))]"/>
```
Comment (i-vyshnevska) [inline mgmtsystem/security/mgmtsystem_security.xml:20]:
```suggestion
        <field name="implied_ids" eval="[(4,ref('group_mgmtsystem_viewer'))]"/>
```

File: mgmtsystem/security/mgmtsystem_security.xml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    </record>
+
+    <record id="group_mgmtsystem_user_manager" model="res.groups">
+        <field name="name">Approving User</field>
+        <field name="comment">Module user with additional permission for approvals.</field>
+        <field name="category_id" ref="module_category_management_system"/>
+        <field name="implied_ids" eval="[(4,ref('mgmtsystem.group_mgmtsystem_user'))]"/>
```
Comment (i-vyshnevska) [inline mgmtsystem/security/mgmtsystem_security.xml:27]:
```suggestion
        <field name="implied_ids" eval="[(4,ref('group_mgmtsystem_user'))]"/>
```

File: mgmtsystem/security/mgmtsystem_security.xml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        <field name="users" eval="[(4, ref('base.user_root')), (4, ref('base.user_admin'))]"/>
+    </record>
+
+    <record id="group_mgmtsystem_auditor" model="res.groups">
+        <field name="name">Auditor</field>
+        <field name="comment">Module user with additional permission specific for the auditor job, such as effectiveness reviews.</field>
+        <field name="category_id" ref="module_category_management_system"/>
+        <field name="implied_ids" eval="[(4,ref('mgmtsystem.group_mgmtsystem_user'))]"/>
```
Comment (i-vyshnevska) [inline mgmtsystem/security/mgmtsystem_security.xml:35]:
```suggestion
        <field name="implied_ids" eval="[(4,ref('group_mgmtsystem_user'))]"/>
```

**Review by i-vyshnevska — CHANGES_REQUESTED** [review]

File: mgmtsystem/models/res_config.py [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    module_mgmtsystem_audit = fields.Boolean(
+        'Audits',
+        help='Provide audit tools.\n'
+        '-This installs the module mgmtsystem_audit.'
+    )
+
+    module_mgmtsystem_review = fields.Boolean(
+        'Reviews',
+        help='Provide review tools.\n'
+        '-This installs the module mgmtsystem_review.'
```
Comment (gurneyalex) [inline mgmtsystem/models/res_config.py]:
```suggestion
        '- This installs the module mgmtsystem_review.'
```

File: mgmtsystem/models/res_config.py [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+    module_mgmtsystem_claim = fields.Boolean(
+        'Claims',
+        help='Provide claim tools.\n'
+        '-This installs the module mgmtsystem_claim.'
+    )
+
+    module_mgmtsystem_audit = fields.Boolean(
+        'Audits',
+        help='Provide audit tools.\n'
+        '-This installs the module mgmtsystem_audit.'
```
Comment (gurneyalex) [inline mgmtsystem/models/res_config.py]:
```suggestion
        '- This installs the module mgmtsystem_audit.'
```

File: mgmtsystem/models/res_config.py [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+    module_mgmtsystem_nonconformity = fields.Boolean(
+        'Nonconformities',
+        help='Provide non conformity tools.\n'
+        '-This installs the module mgmtsystem_nonconformity.'
+    )
+
+    module_mgmtsystem_claim = fields.Boolean(
+        'Claims',
+        help='Provide claim tools.\n'
+        '-This installs the module mgmtsystem_claim.'
```
Comment (gurneyalex) [inline mgmtsystem/models/res_config.py]:
```suggestion
        '- This installs the module mgmtsystem_claim.'
```

File: mgmtsystem/models/res_config.py [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+    module_mgmtsystem_action = fields.Boolean(
+        'Actions',
+        help='Provide actions and improvement opportunities tools.\n'
+        '-This installs the module mgmtsystem_action.'
+    )
+
+    module_mgmtsystem_nonconformity = fields.Boolean(
+        'Nonconformities',
+        help='Provide non conformity tools.\n'
+        '-This installs the module mgmtsystem_nonconformity.'
```
Comment (gurneyalex) [inline mgmtsystem/models/res_config.py]:
```suggestion
        '- This installs the module mgmtsystem_nonconformity.'
```

File: mgmtsystem/models/res_config.py [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+        help='Provide information security tools.\n'
+        '-This installs the module mgmtsystem_information_security.'
+    )
+
+    # Applications
+
+    module_mgmtsystem_action = fields.Boolean(
+        'Actions',
+        help='Provide actions and improvement opportunities tools.\n'
+        '-This installs the module mgmtsystem_action.'
```
Comment (gurneyalex) [inline mgmtsystem/models/res_config.py]:
```suggestion
        '- This installs the module mgmtsystem_action.' 
```

File: mgmtsystem/models/res_config.py [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+    module_mgmtsystem_health_safety = fields.Boolean(
+        'Hygiene & Safety',
+        help='Provide health and safety management tools.\n'
+        '-This installs the module mgmtsystem_health_safety.'
+    )
+
+    module_mgmtsystem_information_security = fields.Boolean(
+        'Information Security',
+        help='Provide information security tools.\n'
+        '-This installs the module mgmtsystem_information_security.'
```
Comment (gurneyalex) [inline mgmtsystem/models/res_config.py]:
```suggestion
        '- This installs the module mgmtsystem_information_security.'
```

File: mgmtsystem/models/res_config.py [hunk 14]
```diff
@@ -0,0 +1,126 @@
...
+    module_mgmtsystem_environment = fields.Boolean(
+        'Environment',
+        help='Provide environment management tools.\n'
+        '-This installs the module mgmtsystem_environment.'
+    )
+
+    module_mgmtsystem_health_safety = fields.Boolean(
+        'Hygiene & Safety',
+        help='Provide health and safety management tools.\n'
+        '-This installs the module mgmtsystem_health_safety.'
```
Comment (gurneyalex) [inline mgmtsystem/models/res_config.py]:
```suggestion
        '- This installs the module mgmtsystem_health_safety.'
```

File: mgmtsystem/models/res_config.py [hunk 15]
```diff
@@ -0,0 +1,126 @@
...
+    module_mgmtsystem_quality = fields.Boolean(
+        'Quality',
+        help='Provide quality management tools.\n'
+        '-This installs the module mgmtsystem_quality.'
+    )
+
+    module_mgmtsystem_environment = fields.Boolean(
+        'Environment',
+        help='Provide environment management tools.\n'
+        '-This installs the module mgmtsystem_environment.'
```
Comment (gurneyalex) [inline mgmtsystem/models/res_config.py]:
```suggestion
        '- This installs the module mgmtsystem_environment.'
```

File: mgmtsystem/models/res_config.py [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+    """This class in needed to activate management system Applications."""
+
+    _inherit = 'res.config.settings'
+
+    # Systems
+
+    module_mgmtsystem_quality = fields.Boolean(
+        'Quality',
+        help='Provide quality management tools.\n'
+        '-This installs the module mgmtsystem_quality.'
```
Comment (gurneyalex) [inline mgmtsystem/models/res_config.py]:
```suggestion
        '- This installs the module mgmtsystem_quality.'
```

File: mgmtsystem/models/res_config.py [hunk 17]
```diff
@@ -0,0 +1,126 @@
+# Copyright (C) 2004-2012 OpenERP S.A. (<http://openerp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields
+
+
+class MgmtsystemConfigSettings(models.TransientModel):
+    """This class in needed to activate management system Applications."""
```
Comment (gurneyalex) [inline mgmtsystem/models/res_config.py]:
```suggestion
    """This class is used to activate management system applications."""
```

File: mgmtsystem/models/res_config.py [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+    module_mgmtsystem_security_event = fields.Boolean(
+        'Feared Events',
+        help='Provide Feared Events.\n'
+        '- This installs the module mgmtsystem_security_event.'
+    )
+
+    module_document_page_approval = fields.Boolean(
+        'Document Page Approval',
+        help='Provide document approval and history. \n'
+        '-This installs the module document_page_approval.'
```
Comment (gurneyalex) [inline mgmtsystem/models/res_config.py]:
```suggestion
        '- This installs the module document_page_approval.'
```

**Review by gurneyalex — APPROVED** [review]

File: mgmtsystem/views/res_config.xml (hunk 4 above)
Comment (simahawk) [reply mgmtsystem/views/res_config.xml:15]: this is wrong, in views you have to use the xid

**Review by jgrandguillaume — APPROVED** [review]
LGMT. Might be we want to add yourself as contributor now ;)

**Comment by simahawk** [comment]
CR addressed, old commits squashed by author and scope and wording fixed up.

**Comment by simahawk** [comment]
WARNING: wait to merge, the cleanup of history makes too hard to migrate each modules by itself, is going to be an hell for every subsequent PR because all the modules' history are quite tied to each other.
I rollback the squashing and cherry-pick only my change.

**Comment by simahawk** [comment]
rolled back. If we want a clean history for 12 we have to do it when all modules are migrated from 11.0.
When it gets green we can merge. /cc @gurneyalex

**Review by i-vyshnevska — APPROVED** [review]
if you still there you can drop pot file

**Pull request merged**